components/HTTP/src/Websocket.cpp

```diff
@@ -0,0 +1,303 @@
+#include "CheckConnections.h"
+#include <esp_log.h>
+#include <iostream>
+static const char* TAG = "Websocket";
+
+
+Websocket::Websocket()
+{
+    static CheckConnections g_check_connections;
+    g_check_connections.start(NULL);
+}
+
+void Websocket::onReceive(httpd_ws_frame_t ws_pkt, httpd_req_t* req)
+{
+    req_handler = req;
+    int type = getTypeMessage(ws_pkt.payload);
+    WebsocketType r = static_cast<WebsocketType>(type);
+    client_data_t client;
+    esp_err_t err;
+    uid_message_t client_uid;
+    const char* null_to_uid = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+    bool update = false;
+
+    switch (r) {
+    case WebsocketType::handshake:
+        ESP_LOGI(TAG, "!!!!!hanshake");
+        err = getClientData(ws_pkt.payload, &client);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "type must be numeric 0 ");
+        };
+
+        if (m_client.size() > 0) {
+            int size = m_client.size();
+            for (size_t i = 0; i < size; i++) {
+                if (memcmp(m_client[i].shaUID, client.shaUID, sizeof(client.shaUID)) == 0) {
+                    m_client[i].fd = client.fd;
+                    update = true;
+                }
+            }
+            if (update == false) {
+                m_client.push_back(client);
+            }
+
+            update = false;
+        } else {
+            m_client.push_back(client);
+        }
+
+        std::cout << "vector: " << m_client.size() << std::endl;
+        break;
+    case WebsocketType::msg:
+        ESP_LOGI(TAG, "!!!!!msg");
+
+        err = messageRecipient(ws_pkt.payload, &client_uid, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error getting message uids %d", err);
+            return;
+        };
+
+        err = sendWsData(client_uid, ws_pkt, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error sending message %d", err);
+            return;
+        };
+
+        break;
+    case WebsocketType::status:
+        ESP_LOGI(TAG, "!!!!!status");
+
+        err = messageRecipient(ws_pkt.payload, &client_uid, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error getting message uids %d", err);
+            return;
+        };
+
+        err = sendWsData(client_uid, ws_pkt, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error sending message %d", err);
+            return;
+        };
+
+        break;
+    case WebsocketType::action:
+        ESP_LOGI(TAG, "!!!!!action");
+
+        err = messageRecipient(ws_pkt.payload, &client_uid, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error getting message uids %d", err);
+            return;
+        };
+
+        err = sendWsData(client_uid, ws_pkt, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error sending message %d", err);
+            return;
+        };
+
+        break;
+    default:
+        break;
```

Comment:
here a default case is mandatory 
```
    default:
        ESP_LOGE(TAG, "Unknow message type");
        break;
```
it's possible to reach the end of the funcion without any "WebsocketType" detected, this error must be handled.
```
    default:
        ESP_LOGE(TAG, "Unknow message type");
        // Do something in case of unknow  msg-type
        break;
```